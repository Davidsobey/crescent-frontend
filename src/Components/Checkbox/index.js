import React from 'react';
import styled from 'styled-components';
import { Checkbox } from 'redux-form-material-ui';
import PropTypes from 'prop-types';

const CheckboxWrapper = styled.div`
  & input {
    width: 400px;
  }
`;

function ThemedCheckbox({ theme, ...rest }) {
  return (
    <CheckboxWrapper>
      <Checkbox {...rest} />
    </CheckboxWrapper>
  );
}

ThemedCheckbox.propTypes = {
  theme: PropTypes.object,
};
export default ThemedCheckbox;
