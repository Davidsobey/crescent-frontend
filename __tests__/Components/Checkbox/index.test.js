import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import Checkbox from '../../../src/Components/Checkbox/index.js';


const mockStore = configureStore();
const initialState = {};
const store = mockStore(initialState);

describe('<Checkbox />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Checkbox store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});