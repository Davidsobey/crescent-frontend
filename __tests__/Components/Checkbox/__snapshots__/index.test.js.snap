// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> render() renders the component 1`] = `
<div
  className="sc-bdVaJa jySyCm"
>
  <ReduxFormMaterialUIWithStyles
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</div>
`;
